src/fu-engine-config.c

```diff
@@ -338,9 +339,23 @@ fu_engine_config_get_show_device_private(FuEngineConfig *self)
 					FU_DAEMON_CONFIG_DEFAULT_SHOW_DEVICE_PRIVATE);
 }
 
+gboolean
+fu_engine_config_get_test_devices(FuEngineConfig *self)
+{
+	return fu_config_get_value_bool(FU_CONFIG(self),
+					"fwupd",
+					"TestDevices",
+					FU_DAEMON_CONFIG_DEFAULT_TEST_DEVICES);
+}
+
 gboolean
 fu_engine_config_get_allow_emulation(FuEngineConfig *self)
 {
+	if (fu_engine_config_get_test_devices(self)) {
+		g_info("allowing emulation as TestDevices is enabled");
+		return TRUE;
+	}
```

Comment:
on the logic? seems a bit orthogonal, no?

Reply:
Well the fwupd.sh tests run all sorts of emulation device tests, which seemed to fit the bill for "virtual devices"

Reply:
Seems like a different thing to me.

Reply:
If you really think they shouldn't be linked; then I think fwupd.sh needs `fwupdtool modify-config AllowEmulation true`

Reply:
I think it's confusing to link test-devices and emulation -- they're two completely different things, and enabling the former isn't obvious that the latter is also enabled.

Reply:
OK; look at the fixup commit.  Does that make more sense to you?  If that passes CI I'll squash it.